### PR TITLE
Update jq to 1.7 (with arm64 support)

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: make/go/dep_jq.mk [hunk 1]
```diff
@@ -9,28 +9,30 @@ $(call _assert_var,CACHE_VERSIONS)
...
 
-# jq does not have an ARM release on Github so we'll use
-# the amd64 version via Rosetta
 ifeq ($(UNAME_OS),Darwin)
-JQ_OS := osx
-JQ_ARCH := -amd64
+JQ_OS := macos
+ifeq ($(UNAME_ARCH),arm64)
+JQ_ARCH := arm64
+else
```
Comment (bufdev) [inline make/go/dep_jq.mk:19]: Look at protoc, we havent done `else` in this case if I remember correctly

File: make/go/dep_jq.mk [hunk 2]
```diff
@@ -9,28 +9,30 @@ $(call _assert_var,CACHE_VERSIONS)
...
-JQ_ARCH := 64
+JQ_ARCH := amd64
 endif
 endif
 JQ := $(CACHE_VERSIONS)/jq/$(JQ_VERSION)
 $(JQ):
 	@rm -f $(CACHE_BIN)/jq
 	@mkdir -p $(CACHE_BIN)
 	curl -sSL \
-		https://github.com/stedolan/jq/releases/download/jq-$(JQ_VERSION)/jq-$(JQ_OS)$(JQ_ARCH) \
```
Comment (mfridman) [inline make/go/dep_jq.mk:33]: Oh, didn't realized this moved.

File: make/go/dep_jq.mk (hunk 1 above)
Comment (pkwarren) [reply make/go/dep_jq.mk]: Pushed an update - is that what you meant?

File: make/go/dep_jq.mk [hunk 3]
```diff
@@ -9,28 +9,30 @@ $(call _assert_var,CACHE_VERSIONS)
...
-# jq does not have an ARM release on Github so we'll use
-# the amd64 version via Rosetta
 ifeq ($(UNAME_OS),Darwin)
-JQ_OS := osx
-JQ_ARCH := -amd64
+JQ_OS := macos
+ifeq ($(UNAME_ARCH),x86_64)
+JQ_ARCH := amd64
+else
+JQ_ARCH := $(UNAME_ARCH)
```
Comment (mfridman) [inline make/go/dep_jq.mk:20]: Should this just be `arm64` ?

Comment (pkwarren) [reply make/go/dep_jq.mk:20]: That's how it was before - I changed it to match this pattern in the protoc file: https://github.com/bufbuild/makego/blob/99dd87fd8e550f6875c3a83b18ec601db8c0aadc/make/go/dep_protoc.mk#L27C16-L27C29

**Review by mfridman — APPROVED** [review]

**Pull request merged**